index.js

```diff
@@ -365,11 +366,11 @@ module.exports = class NoiseSecretStream extends Duplex {
   }
 
   _setupSecretStream (tx, rx, handshakeHash, publicKey, remotePublicKey) {
-    const buf = b4a.allocUnsafe(3 + IDHEADERBYTES)
+    const buf = b4a.allocUnsafeSlow(3 + IDHEADERBYTES)
```

Comment:
where is this retained?

Reply:
On line 372, the `buf.subarray(3 +32)` (which becomes `this._encrypt.header`.

It's easier to see when the dependency to sodium-secretstream is in, because then the tests pass. Removing the allocUnsafeSlow then shows where it fails with that gone.

Note: we could also unslab the subarray on line 372, but that felt a bit silly for the 3 byte difference.

Reply:
Edit: it's actually a 35 byte difference, so maybe there is an argument for unslabbing there, particularly if the code then becomes more readable

Reply:
no need to save 35 bytes. i see it now, makes sense